homeassistant/components/opentherm_gw/sensor.py

```diff
@@ -1,20 +1,621 @@
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement="starts",
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_DHW_PUMP_STARTS,
+            friendly_name_format="Hot Water Pump Starts {}",
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement="starts",
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_DHW_BURNER_STARTS,
+            friendly_name_format="Hot Water Burner Starts {}",
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement="starts",
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_TOTAL_BURNER_HOURS,
+            friendly_name_format="Total Burner Hours {}",
+            device_class=SensorDeviceClass.DURATION,
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement=UnitOfTime.HOURS,
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_CH_PUMP_HOURS,
+            friendly_name_format="Central Heating Pump Hours {}",
+            device_class=SensorDeviceClass.DURATION,
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement=UnitOfTime.HOURS,
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_DHW_PUMP_HOURS,
+            friendly_name_format="Hot Water Pump Hours {}",
+            device_class=SensorDeviceClass.DURATION,
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement=UnitOfTime.HOURS,
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_DHW_BURNER_HOURS,
+            friendly_name_format="Hot Water Burner Hours {}",
+            device_class=SensorDeviceClass.DURATION,
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement=UnitOfTime.HOURS,
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_MASTER_OT_VERSION,
+            friendly_name_format="Thermostat OpenTherm Version {}",
+            suggested_display_precision=SENSOR_FLOAT_SUGGESTED_DISPLAY_PRECISION,
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_SLAVE_OT_VERSION,
+            friendly_name_format="Boiler OpenTherm Version {}",
+            suggested_display_precision=SENSOR_FLOAT_SUGGESTED_DISPLAY_PRECISION,
+        ),
+    ),
```

Comment:
Why does a version have a display precision?

Are the thermostat and the boiler 2 separate devices? Should we split them up and use the device_info `sw_version` for this?

Reply:
The display precision is there because the version is reported in the protocol as a `float`. OpenTherm floats are only accurate to 1/256th, so the precision is there to make sure a value of `0.1` doesn't show as `0,1015625` or `0,09765625`.

The thermostat and boiler are indeed 2 different devices, the OpenTherm Gateway sits in between them.
I could separate them into 3 devices, but as support for a lot of features is optional and differ hugely between different manufacturers and models, most of them would end up with loads of `unknown` values. Neither my thermostat nor my boiler report their OpenTherm Version, for example.

If it is still considered a good idea to split these devices, then I suggest to make it a separate PR as that operation will be a bit more involved.

Reply:
Oh yea definitely, I just picked this PR to leave comments because I can leave comments for every sensor. I think it would be nice to create separate devices for them, also because in the end we want to move to `_attr_has_entity_name = True`, which means that naming is going to change to `{device} {entity name}`. So if we could have a device for a boiler, thermostat and otgw, we can make it nicer and also allow the user to place the device in the right area. (as I can imagine you probably have your thermostat in the living room, but your boiler in the attic)

Reply:
Like, every comment I leave here is not a blocker, it's just asking the right questions to give hints and a bit of direction to how we can make it even better

Reply:
Sounds like the way to go is different devices. So far I have used the `source` info in the friendly name and entity_id to differentiate between values known by the thermostat and boiler respectively (they are not always the same), but different devices could make that separation clearer. There has actually already been some confusion about this that lead to a bug report in the past.